src/types/qwxdgshell.cpp

```diff
@@ -280,6 +282,13 @@ void QWXdgSurface::forEachPopupSurface(wlr_surface_iterator_func_t iterator, voi
     wlr_xdg_surface_for_each_popup_surface(handle(), iterator, userData);
 }
 
+#if WLR_VERSION_MINOR == 16
+bool QWXdgSurface::isXdgSurface(QWSurface *surface)
+{
+    return wlr_surface_is_xdg_surface(surface->handle());
```

Comment:
If the function is droped on wlroots 0.17, we don't provide it on public interface of QWlroots.
You can direct using `wlr_surface_is_xdg_surface` at `QWXdgSurface::from`.